### ISSUE TITLE
Pylint 3.0.1 | invalid-field-call | false positive

### ISSUE THREAD

**Issue by LefterisJP** [pylint-dev/pylint#9130]
### Bug description

```python
Using a `Field` in a child class of a dataclass seems to trigger `invalid-field-call`



from dataclasses import dataclass, field


@dataclass(init=True, repr=False, eq=False, order=False, unsafe_hash=False, frozen=True)
class Parent():
    name: str = field(init=False)

class Child(Parent):
    age: int = field(init=False)
```

### Configuration

```ini
disable=all
enable=invalid-field-call
```

### Command used

```shell
pylint test_format.py
```

### Pylint output

```python
************* Module test_format
test_format.py:9:15: E3701: Invalid usage of field(), it should be used within a dataclass or the make_dataclass() function. (invalid-field-call)
```

### Expected behavior

I would expect no errors since it's inside a dataclass by inheritance

### Pylint version

```shell
pylint 3.0.1
astroid 3.0.0
Python 3.10.12 (main, Jul  2 2023, 10:24:23) [GCC 13.1.1 20230429]
```

### OS / Environment

Archlinux
